### DISCUSSION TITLE
Update GoogleToolboxForMac

### DISCUSSION THREAD

**Discussion by emmaB93** [capawesome-team/capacitor-mlkit#157] [General]
The library is using GoogleToolboxForMac, correct?

Could you please use the latest version which includes the missing privacy infos?
I'm currently getting the following message:

Descriptions for the following required API reason(s) may be missing: 1
[0] NSPrivacyAccessedAPICategoryUserDefaults

While analyzing my project I'm getting the following warnings:

`If the directory you are analyzing is the app project directory, your app's privacy manifest may be affected by these analysis results: 3
[0] NSPrivacyAccessedAPICategoryFileTimestamp:stat,NSURLContentModificationDateKey:/App/Pods/MLKitCommon/Frameworks/MLKitCommon.framework/MLKitCommon
[1] NSPrivacyAccessedAPICategoryDiskSpace:NSFileSystemFreeSize:/App/Pods/MLKitCommon/Frameworks/MLKitCommon.framework/MLKitCommon
[2] NSPrivacyAccessedAPICategoryUserDefaults:NSUserDefaults:/App/Pods/MLKitCommon/Frameworks/MLKitCommon.framework/MLKitCommon` 

I'm not 100% sure if this is caused by the capacitor plugin? (I use the latest version "@capacitor-mlkit/barcode-scanning": "6.1.0")
Thanks for helping!

**Comment by robingenz** [comment]
We already use the latest version of the `GoogleMLKit/BarcodeScanning`, see: https://github.com/capawesome-team/capacitor-mlkit/blob/9190bc86a5be0f5f4f47526014c11e8df5c0e10e/packages/barcode-scanning/ios/Podfile#L12

Here you can find a list of the latest versions: https://developers.google.com/ml-kit/release-notes

**EDIT**: I was wrong. We are not yet using the latest version, see https://github.com/capawesome-team/capacitor-mlkit/pull/179.

**Comment by rye-welz-geselowitz** [comment]
I am also struggling with the missing privacy manifest, and would love advice! 

Upon submission to the App Store, I received the following warning from App Store Connect:

> Missing privacy manifest - Your app includes “Frameworks/GoogleToolboxForMac.framework/GoogleToolboxForMac”, which includes GoogleToolboxForMac, an SDK that was identified in the documentation as a privacy-impacting third-party SDK.


I am also using `@capacitor-mlkit/barcode-scanning` 6.1.0. I understand from my `Podfile.lock` that  `GoogleToolboxForMac` is  included in my project through `CapacitorMlkitBarcodeScanning` **[1]** . 

I understand that `capacitor-mlkit/barcode-scanning` is using the latest version of `GoogleMLKit/BarcodeScanning`, so there's nothing the  `capacitor-mlkit/barcode-scanning`  maintainers need to change. 

But my questions are:
1. Is my summary of the situation correct?: _`capacitor-mlkit/barcode-scanning` uses the latest version of `GoogleMLKit/BarcodeScanning`, but `GoogleMLKit/BarcodeScanning` includes (several layers down) a dependency on a version of `GoogleToolboxForMac` that does not include the required privacy manifest._
2. What steps (if any) can I take to decrease the likelihood of my iOS app, which relies on `capacitor-mlkit/barcode-scanning`, being rejected from the App Store on this basis?

I love this plugin and I want to make sure I can continue using it! Apologies if this is the wrong place for these questions, please feel free to direct me elsewhere. :) 

------
**[1]** 
```
PODS:
  - AparajitaCapacitorSecureStorage (6.0.1):
    - Capacitor
    - KeychainSwift (~> 21.0)
  - Capacitor (6.1.1):
    - CapacitorCordova
  - CapacitorApp (6.0.1):
    - Capacitor
  - CapacitorCommunitySqlite (6.0.1):
    - Capacitor
    - SQLCipher
    - ZIPFoundation
  - CapacitorCordova (6.1.1)
  - CapacitorDevice (6.0.0):
    - Capacitor
  - CapacitorLiveUpdates (0.3.1):
    - Capacitor
    - IonicLiveUpdates (~> 0.5.4)
  - CapacitorMlkitBarcodeScanning (6.1.0):
    - Capacitor
    - GoogleMLKit/BarcodeScanning (= 5.0.0)
  - CapacitorNetwork (6.0.1):
    - Capacitor
  - CapacitorPreferences (6.0.1):
    - Capacitor
  - CapacitorSplashScreen (6.0.2):
    - Capacitor
  - GoogleDataTransport (9.4.1):
    - GoogleUtilities/Environment (~> 7.7)
    - nanopb (< 2.30911.0, >= 2.30908.0)
    - PromisesObjC (< 3.0, >= 1.2)
  - GoogleMLKit/BarcodeScanning (5.0.0):
    - GoogleMLKit/MLKitCore
    - MLKitBarcodeScanning (~> 4.0.0)
  - GoogleMLKit/MLKitCore (5.0.0):
    - MLKitCommon (~> 10.0.0)
  - GoogleToolboxForMac/DebugUtils (2.3.2):
    - GoogleToolboxForMac/Defines (= 2.3.2)
  - GoogleToolboxForMac/Defines (2.3.2)
  - GoogleToolboxForMac/Logger (2.3.2):
    - GoogleToolboxForMac/Defines (= 2.3.2)
  - "GoogleToolboxForMac/NSData+zlib (2.3.2)":
    - GoogleToolboxForMac/Defines (= 2.3.2)
  - "GoogleToolboxForMac/NSDictionary+URLArguments (2.3.2)":
    - GoogleToolboxForMac/DebugUtils (= 2.3.2)
    - GoogleToolboxForMac/Defines (= 2.3.2)
    - "GoogleToolboxForMac/NSString+URLArguments (= 2.3.2)"
  - "GoogleToolboxForMac/NSString+URLArguments (2.3.2)"
  - GoogleUtilities/Environment (7.13.3):
    - GoogleUtilities/Privacy
    - PromisesObjC (< 3.0, >= 1.2)
  - GoogleUtilities/Logger (7.13.3):
    - GoogleUtilities/Environment
    - GoogleUtilities/Privacy
  - GoogleUtilities/Privacy (7.13.3)
  - GoogleUtilities/UserDefaults (7.13.3):
    - GoogleUtilities/Logger
    - GoogleUtilities/Privacy
  - GoogleUtilitiesComponents (1.1.0):
    - GoogleUtilities/Logger
  - GTMSessionFetcher/Core (3.5.0)
  - IonicLiveUpdates (0.5.5)
  - KeychainSwift (21.0.0)
  - MLImage (1.0.0-beta5)
  - MLKitBarcodeScanning (4.0.0):
    - MLKitCommon (~> 10.0)
    - MLKitVision (~> 6.0)
  - MLKitCommon (10.0.0):
    - GoogleDataTransport (~> 9.0)
    - GoogleToolboxForMac/Logger (~> 2.1)
    - "GoogleToolboxForMac/NSData+zlib (~> 2.1)"
    - "GoogleToolboxForMac/NSDictionary+URLArguments (~> 2.1)"
    - GoogleUtilities/UserDefaults (~> 7.0)
    - GoogleUtilitiesComponents (~> 1.0)
    - GTMSessionFetcher/Core (< 4.0, >= 1.1)
  - MLKitVision (6.0.0):
    - GoogleToolboxForMac/Logger (~> 2.1)
    - "GoogleToolboxForMac/NSData+zlib (~> 2.1)"
    - GTMSessionFetcher/Core (< 4.0, >= 1.1)
    - MLImage (= 1.0.0-beta5)
    - MLKitCommon (~> 10.0)
  - nanopb (2.30910.0):
    - nanopb/decode (= 2.30910.0)
    - nanopb/encode (= 2.30910.0)
  - nanopb/decode (2.30910.0)
  - nanopb/encode (2.30910.0)
  - PromisesObjC (2.4.0)
  - SQLCipher (4.6.1):
    - SQLCipher/standard (= 4.6.1)
  - SQLCipher/common (4.6.1)
  - SQLCipher/standard (4.6.1):
    - SQLCipher/common
  - ZIPFoundation (0.9.19)
  ```

**Reply by robingenz** [reply]
No, I was wrong. We are not yet using the latest version, see https://github.com/capawesome-team/capacitor-mlkit/pull/179.

**Reply by rye-welz-geselowitz** [reply]
Awesome, thanks for letting me know!

**Comment by PChougule2001** [comment]
hey, how did you upgrade GoogleToolBoxForMac SDK with pods? because i have it with version 2.4.1 and i want to upgrade it to version 4.2.1 or 5.0.0. can you pls tell what step did you perform to upgrade it? in my case pod update is not upgrading this SDK.
Thanks in advance

**Reply by robingenz** [reply]
Make sure you have the latest plugin version installed, delete the `Podfile.lock` and run `npx cap sync`.

**Reply by PChougule2001** [reply]
Thank you @robingenz, It worked!

**Comment by acatt08** [comment]
hello, i also have a problem with updating GoogleToolBoxForMac. I have capacitor version 6.0.0. I have CapacitorMlkitBarcodeScanning version 6.2.0. I order to update it to 7.0.0 i need to update capacitor from version ^6.0.0 to ^7.0.0, is that right?

**Reply by robingenz** [reply]
Yes, that's correct.

**Reply by acatt08** [reply]
Thanks @robingenz Do you maybe know if there is alternative for fixing Missing privacy manifest for GoogleToolboxForMac(old version doesn't have xcprivacy.privacy)? It's a big project and updating capacitor would be a hard work

**Reply by robingenz** [reply]
Maybe this helps you: https://forum.ionicframework.com/t/handling-privacy-manifest-requirements-with-outdated-firebase-and-angular-version/246643/10 However, i can't provide any support for this.

**Comment by elzone** [comment]
Hello,
Is it possible to make update for version 6 (@capacitor-mlkit/barcode-scanning) to the version (maybe 6.3.0) with correct privacy manifest for GoogleToolboxForMac?
I can't update my capacitor app to version 7 for now, but need update app in AppStore, but get this message: 
ITMS-91061: Missing privacy manifest - Your app includes “Frameworks/GoogleToolboxForMac.framework/GoogleToolboxForMac”, which includes GoogleToolboxForMac, an SDK that was identified in the documentation as a commonly used third-party SDK. If a new app includes a commonly used third-party SDK, or an app update adds a new commonly used third-party SDK, the SDK must include a privacy manifest file. Please contact the provider of the SDK that includes this file to get an updated SDK version with a privacy manifest.

You will help me with this update of the barcode-scanning plugin.
Thank you.

**Reply by acatt08** [reply]
I fixed it by following steps from https://forum.ionicframework.com/t/handling-privacy-manifest-requirements-with-outdated-firebase-and-angular-version/246643/10

**Reply by elzone** [reply]
did you put it like this?
![image](https://github.com/user-attachments/assets/a062033a-defd-45a8-be05-4a5192d4f12e)

**Reply by elzone** [reply]
sorry, this aproach is not good, cause each time i sync ios project i have to put this file again (or maybe i don't know something). if it will be included into GoogleToolboxForMac will be much better

**Reply by acatt08** [reply]
Yes, i followed instructions from the image above. From what I know, either this or update on capacitor 7

**Reply by elzone** [reply]
i see,
thank you for the info

**Reply by elzone** [reply]
@robingenz , what do you think about 6.3.0 version? is it possible?

**Reply by robingenz** [reply]
@elzone Sorry, i can only provide support for the latest version. I have never tried to create a privacy manifest for a third party SDK.

**Reply by elzone** [reply]
@robingenz , maybe it's possible to include updated version of GoogleToolboxForMac into @capacitor-mlkit/barcode-scanning v6, like you did in v7?

**Reply by robingenz** [reply]
@elzone We don't have a direct dependency to `GoogleToolboxForMac`. The only dependency besides Capacitor itself is `GoogleMLKit/BarcodeScanning`, see https://github.com/capawesome-team/capacitor-mlkit/blob/ce9e35e43c40b82569dd180d1180317196181f29/packages/barcode-scanning/CapacitorMlkitBarcodeScanning.podspec#L16.

**Reply by elzone** [reply]
i see,
thank you for the info
